x/dispensation/keeper/grpc_querier.go

```diff
@@ -33,30 +31,20 @@ func (q Querier) AllDistributions(ctx context.Context,
 
 func (q Querier) RecordsByDistributionName(ctx context.Context, request *types.QueryRecordsByDistributionNameRequest) (*types.QueryRecordsByDistributionNameResponse, error) {
 	records := new(types.DistributionRecords)
-	switch request.Status {
-	case types.ClaimStatus_CLAIM_STATUS_PENDING:
-		*records = q.keeper.GetRecordsForNamePending(sdk.UnwrapSDKContext(ctx), request.DistributionName)
-	case types.ClaimStatus_CLAIM_STATUS_COMPLETED:
-		*records = q.keeper.GetRecordsForNameCompleted(sdk.UnwrapSDKContext(ctx), request.DistributionName)
-	default:
-		*records = q.keeper.GetRecordsForNameAll(sdk.UnwrapSDKContext(ctx), request.DistributionName)
+	*records = q.keeper.GetRecordsForName(sdk.UnwrapSDKContext(ctx), request.DistributionName, request.Status)
+	if request.Status == types.DistributionStatus_DISTRIBUTION_STATUS_UNSPECIFIED {
```

Comment:
Why are the pending records being added if request status is unspecified?

Reply:
User can either specify , Pending Completed , or anything else , I am assuming the user does 
records-by-name [distribution name].  , which I am assuming they are not interested in the status .